### PR TITLE
Drivetrain

### PR DESCRIPTION
**Pull request by continuumsrc**

### CODE REVIEW

File: src/main/deploy/pathplanner/navgrid.json
Comment (garrettwitowski) [inline src/main/deploy/pathplanner/navgrid.json:1]: What is this?

File: src/main/deploy/pathplanner/paths/New Path.path
Comment (garrettwitowski) [inline src/main/deploy/pathplanner/paths/New Path.path:1]: Have we tested this and verified it works?

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/drive/Drivetrain.java:52]: Why are we removing the odometry? Don't we need the odometry in addition to the estimator?

Comment (Irishumanoid) [reply src/main/java/frc/robot/subsystems/drive/Drivetrain.java:52]: Either way, we need odometry for path planning, so please add it back

File: src/main/deploy/pathplanner/paths/New Path.path
Comment (EmDash00) [reply src/main/deploy/pathplanner/paths/New Path.path:1]: No. Not sure what it is. Got committed but we need the vision additions from drivetrain.

File: src/main/deploy/pathplanner/navgrid.json
Comment (EmDash00) [reply src/main/deploy/pathplanner/navgrid.json:1]: Something created by pathplanner.

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java
Comment (continuumsrc) [reply src/main/java/frc/robot/subsystems/drive/Drivetrain.java:52]: I don't think we need odometry, the path planner seems to only need pose2d which the estimator provides, its a wrapper around odometry.

Comment (Irishumanoid) [reply src/main/java/frc/robot/subsystems/drive/Drivetrain.java:52]: That should be fine then. Please also update the swerve drive pose estimator in periodic with the addVisionMeasurement from the photon pose estimator from the vision subsystem. You should incorporate this into a separate PR (it's not urgent).

**Review by Irishumanoid — DISMISSED** [review]

**Review by Irishumanoid — COMMENTED** [review]
No more major changes, just get CI to pass before we merge. Run:
`./gradlew SpotlessApply`

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java
Comment (garrettwitowski) [reply src/main/java/frc/robot/subsystems/drive/Drivetrain.java:52]: I would _strongly_ encourage y'all to get clarity on whether or not odometry is necessary before removing it/merging this in. I won't block the merge on this though.

File: src/main/deploy/pathplanner/navgrid.json
Comment (garrettwitowski) [reply src/main/deploy/pathplanner/navgrid.json:1]: Why do we have it? Have we verified its functionality? This is directed at students.

File: src/main/deploy/pathplanner/paths/New Path.path
Comment (garrettwitowski) [reply src/main/deploy/pathplanner/paths/New Path.path:1]: Students, have we verified this works? Also, "New Path" is not a particularly descriptive filename. If this is necessary and functional, please update it to be more descriptive.

**Review by garrettwitowski — CHANGES_REQUESTED** [review]
My recommendation is to get closure on the odometry debate before merging this, but I'll leave that at your discretion. However, please see my comments about the 2 new files.

File: src/main/deploy/pathplanner/navgrid.json
Comment (Irishumanoid) [reply src/main/deploy/pathplanner/navgrid.json:1]: It constructs the field and notable markers for field elements, etc. It's auto generated by the new version of path planner and a lot of other teams use it, so it should be fine.

File: src/main/deploy/pathplanner/paths/New Path.path
Comment (Irishumanoid) [reply src/main/deploy/pathplanner/paths/New Path.path:1]: This can be deleted for now. It's just the default path name from the GUI and shows up once synced to a project.

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java
Comment (Irishumanoid) [reply src/main/java/frc/robot/subsystems/drive/Drivetrain.java:52]: It's not. I checked. The first comment I wrote was a bit hasty.

**Review by Irishumanoid — APPROVED** [review]

**Pull request merged**
